apollo-federation/src/schema/position.rs

```diff
@@ -1087,22 +1077,10 @@ impl ScalarTypeDefinitionPosition {
             .scalar_types
             .contains_key(&self.type_name)
         {
-            return Err(SingleFederationError::Internal {
-                message: format!("Type \"{}\" has not been pre-inserted", self),
-            }
-            .into());
+            bail!(r#"Type "{self}" has not been pre-inserted"#);
         }
         if schema.schema.types.contains_key(&self.type_name) {
-            // TODO: Allow built-in shadowing instead of ignoring them
-            if is_graphql_reserved_name(&self.type_name)
```

Comment:
This implies that adding a duplicate reserved name is now also an error. It makes sense in my opinion.